crates/bevy_core_pipeline/src/core_3d/main_pass_3d_node.rs

```diff
@@ -88,7 +100,14 @@ impl Node for MainPass3dNode {
                     // NOTE: The opaque main pass loads the depth buffer and possibly overwrites it
                     depth_ops: Some(Operations {
                         // NOTE: 0.0 is the far plane due to bevy's use of reverse-z projections.
-                        load: camera_3d.depth_load_op.clone().into(),
+                        load: match prepass_settings {
+                            Some(PrepassSettings {
+                                depth_enabled: true,
+                                ..
+                            }) => Camera3dDepthLoadOp::Load,
```

Comment:
we should always load if prepass is enabled, even if we don't output a readable copy of the depth texture

Reply:
I'm not sure I get why. What if someone set the load_op to clear? Should we just ignore it?

Reply:
sorry, i see that you added `PrepassSettings` to the camera bundle. I was assuming presence of the prepass settings was deciding whether it was run or not (as is the case with bloom, fxaa, etc).

if we want it included in the bundle then i guess we would need separate flags for "run" and "output depth texture". i think maybe it's better not to include it in the camera bundle and require people to add it to cameras manually. if we do that then presence of the settings means "run prepass", which has to generate depths (otherwise the prepass output would not be correct due to lack of depth-testing in the prepass itself) and that can be used to reduce overdraw in the main pass. then user can optionally choose to generate a depth texture, normal texture, etc, as well.

i think this is worthwhile since the readable depth texture requires a copy and takes memory, and the depth texture is only really useful if you want to read depth in the main pass, which is probably quite rare. i think most of the features that require it will be in separate passes that run before the main pass anyway (and so can read from the normal depth buffer - the only reason we need to copy it is because in the main pass we can't bind it as a depth buffer and a readable texture simultaneously).

so assuming we don't include PrepassSettings in the bundle, and that presense means "run the prepass", then i think the above would make sense - we would always want to use the generated depth buffer in the main pass. if i missed a decision about including it in the bundle then we'd need separate flags to run it and to output a copy of the depth texture. then also the comment about the prepass always reducing overdraw would be wrong.

Reply:
I need to think about this more, but I prefer having it always in the bundle because it makes it easier to discover. That was the main reason why I included it in the bundle instead of using it's presence as the toggle.

Right now, enabling the prepass is controlled per material on the MaterialPlugin. Changing it to use the presence of the settings as a toggle would make it harder to control it per material I think.

Reply:
> easier to discover

i'm not sure i think it's worthwhile personally, but it's a reasonable view. then i guess you need to choose from
- a separate flag in prepass settings to say whether to run at all, since there could definitely be some views where a prepass isn't required. note that combinations with run=false and others=true don't make sense
- always making a copy of the depth texture. feels wasteful but i'm not sure if it's a real problem
- not using the generated prepass depth info in the main pass if the depth texture copy isn't enabled. seems inefficient as well
- use the generated depth info in the main pass if *any* prepass setting is true. this is a bit magical but at least it does the right thing and shouldn't ever lead to wrong results. it will mean there's no way to get overdraw reduction without requesting at least one prepass output

Reply:
@robtfm so, I switched to having separate components where their presence controls if the prepass runs and associated textures are created, but reading this thread again I'm not entirely sure what I should do here. Right now this still only happens if the depth prepass is enabled, but are you saying that I should use Load if _any_ prepass component is present. As in, even if there's only the NormalPrepass component added to the camera, I should still always load the depth texture?

Reply:
Alright, I think this does everything correctly now. If DepthPrepass or NormalPrepass is present, it will always load the depth buffer.

The copy to texture of the depth buffer only happens when DepthPrepass is present